### PR TITLE
Wrap errors in fragment

### PR DESCRIPTION
**Pull request by alanbernstein**
## Overview

[Describe what this pull request addresses.]

Partially Fixes pilosa/general#28

## Pull request checklist

- [ ] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [ ] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [ ] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [ ] I have resolved any merge conflicts.
- [ ] I have included tests that cover my changes.
- [ ] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

File: fragment.go [hunk 1]
```diff
@@ -518,13 +518,13 @@ func (f *Fragment) SetFieldValue(columnID uint64, bitDepth uint, value uint64) (
 	for i := uint(0); i < bitDepth; i++ {
 		if value&(1<<i) != 0 {
 			if c, err := f.setBit(uint64(i), columnID); err != nil {
-				return changed, err
+				return changed, errors.Wrap(err, "setting")
```
Comment (travisturner) [inline fragment.go]: i'm confused about the level of detail necessary.
wouldn't a wrapped error here look like `setting field value: setting`?
in that case, it seems like this should be `setting bit`
perhaps even better: `"setting bit %d", i`

File: fragment.go [hunk 2]
```diff
@@ -518,13 +518,13 @@ func (f *Fragment) SetFieldValue(columnID uint64, bitDepth uint, value uint64) (
...
 			} else if c {
 				changed = true
 			}
 		} else {
 			if c, err := f.clearBit(uint64(i), columnID); err != nil {
-				return changed, err
+				return changed, errors.Wrap(err, "clearing")
```
Comment (travisturner) [inline fragment.go]: see `setting` above.

File: fragment.go [hunk 3]
```diff
@@ -533,7 +533,7 @@ func (f *Fragment) SetFieldValue(columnID uint64, bitDepth uint, value uint64) (
 
 	// Mark value as set.
 	if c, err := f.setBit(uint64(bitDepth), columnID); err != nil {
-		return changed, err
+		return changed, errors.Wrap(err, "marking")
```
Comment (travisturner) [inline fragment.go]: seems like this should specify that it's the not-null row: `marking not-null`

File: fragment.go [hunk 4]
```diff
@@ -571,10 +571,10 @@ func (f *Fragment) importSetFieldValue(columnID uint64, bitDepth uint, value uin
 	// Mark value as set.
 	p, err := f.pos(uint64(bitDepth), columnID)
 	if err != nil {
-		return changed, err
+		return changed, errors.Wrap(err, "marking")
```
Comment (travisturner) [inline fragment.go]: again, `marking` seems vague

File: fragment.go [hunk 5]
```diff
@@ -1540,7 +1540,7 @@ func (f *Fragment) flushCache() error {
 func (f *Fragment) WriteTo(w io.Writer) (n int64, err error) {
 	// Force cache flush.
 	if err := f.FlushCache(); err != nil {
-		return 0, err
+		return 0, errors.Wrap(err, "flushing")
```
Comment (travisturner) [inline fragment.go]: `flushing cache`?

File: fragment.go (hunk 1 above)
Comment (alanbernstein) [reply fragment.go]: I think your first reaction is right, this is overly detailed

**Pull request merged**